### PR TITLE
Add admin grpc service

### PR DESCRIPTION
**Pull request by 80degreeswest**
Add an admin grpc service to perform various buildfarm management tasks, such as terminate hosts, reboot instances, etc.

### CODE REVIEW

File: src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class AwsAdmin implements Admin {
+  private static final Logger logger = Logger.getLogger(Admin.class.getName());
+  private final AmazonEC2 ec2;
+  private final AWSSimpleSystemsManagement ssm;
+
+  public AwsAdmin() {
+    ec2 = AmazonEC2ClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
```
Comment (80degreeswest) [inline src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java:37]: TODO: Config

File: src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ssm = AWSSimpleSystemsManagementClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
```
Comment (80degreeswest) [inline src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java:38]: TODO: Config

File: src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    logger.info(String.format("Terminated host: %s", hostId));
+  }
+
+  @Override
+  public void stopContainer(String hostId, String containerName) {
+    String stopContainerCmd = "docker ps | grep " + containerName + " | awk '{print $1 }' | xargs -I {} docker stop {}";
+    Map<String, List<String>> parameters = new HashMap<String, List<String>>();
+    parameters.put("commands", Collections.singletonList(stopContainerCmd));
+    ssm.sendCommand(new SendCommandRequest().withDocumentName("AWS-RunShellScript").withInstanceIds(hostId).withParameters(parameters));
+    logger.info(String.format("Stopped container: %s on host: %s", containerName, hostId));
```
Comment (luxe) [inline src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java:53]: should we confirm that the command succeeded, and that the container actually stopped?  I guess buildfarm admin has always just done it this way too.

**Comment by 80degreeswest** [comment]
> I understand the scheduler getting the grpcAdmin endpoints. But how does the awsAdmin come into play? How will its methods get called?

AdminService determines what is called based on the config (deployment_environment)

File: src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java (hunk 3 above)
Comment (80degreeswest) [reply src/main/java/build/buildfarm/common/admin/aws/AwsAdmin.java:53]: Nope, we should just fire off the event and forget about it. We do send the status back so we can listen for errors if we want to on the client side. The client can handle refreshing of the status on the front end.

**Pull request merged**